src/rpc-extension-ws.ts

```diff
@@ -1,15 +1,21 @@
 import { RpcCommon, IPromiseCallbacks } from "./rpc-common";
 import * as WebSocket from "ws";
+import { IChildLogger } from "@vscode-logging/types";
+import { noopLogger } from "./noop-logger";
 
 export class RpcExtensionWebSockets extends RpcCommon {
+  private static readonly className = "RpcExtensionWebSockets";
+  private readonly logger: IChildLogger;
   ws: WebSocket;
 
-  constructor(ws: WebSocket) {
-    super();
+  constructor(ws: WebSocket, logger: IChildLogger = noopLogger) {
+    super(logger.getChildLogger({ label: RpcExtensionWebSockets.className }));
+    this.logger = logger.getChildLogger({ label: RpcExtensionWebSockets.className });
```

Comment:
why didn't you call it 'baseLogger' as in rpc-common?
related issue in yeoman-ui -> https://github.com/SAP/yeoman-ui/issues/516